EventSourcing.Cosmos/CosmosStoredProcedures.cs

```diff
@@ -7,66 +7,67 @@ namespace EventSourcing.Cosmos;
 /// </summary>
 public static class CosmosStoredProcedures
 {
-    private const string BulkDeleteId = "DeleteAggregateAll";
+  private const string BulkDeleteId = "DeleteAggregateAll";
 
-    public static async Task CreateDeleteAggregateAllProcedure(this Container container)
+  public static async Task CreateDeleteAggregateAllProcedure(this Container container)
+  {
+    // Verify code currently stored for DeleteAggregateAll procedure and update only when needed
+    if (await container.VerifyStoredProcedure(BulkDeleteId, StoredProcedures.DeleteAggregateAll))
+      return;
+
+    await container.TryDeleteStoredProcedure(BulkDeleteId);
+    await container.CreateStoredProcedure(BulkDeleteId, StoredProcedures.DeleteAggregateAll);
+  }
+
+  public static async Task<int> ExecuteDeleteAggregateAllProcedure(this Container container, Guid partitionId,
+    Guid aggregateId)
+  {
+    var deleted = 0;
+    DeleteAggregateAllResponse response;
+
+    do
     {
-        // Verify code currently stored for DeleteAggregateAll procedure and update only when needed
-        if (await container.VerifyStoredProcedure(BulkDeleteId, StoredProcedures.DeleteAggregateAll))
-            return;
-        
-        await container.TryDeleteStoredProcedure(BulkDeleteId);
-        await container.CreateStoredProcedure(BulkDeleteId, StoredProcedures.DeleteAggregateAll);
-    }
-    
-    public static async Task<int> ExecuteDeleteAggregateAllProcedure(this Container container, Guid partitionId, Guid aggregateId)
+      response = (await container.Scripts.ExecuteStoredProcedureAsync<DeleteAggregateAllResponse>(
+        BulkDeleteId,
+        new PartitionKey($"{partitionId}"),
+        new dynamic[] { container.Id, partitionId.ToString(), aggregateId.ToString() })).Resource;
```

Comment:
The `PartitionKey` & `dynamic[]` could be instantiated prior to the `do` `while` loop